app/services/strategy_marketplace_service.py

```
@@ -1756,18 +1769,98 @@ async def _get_user_strategy_portfolio_impl(self, user_id: str) -> Dict[str, Any
+        """Rebuild the user's strategy portfolio directly from the database."""
+
+        try:
+            import uuid
+
+            try:
+                user_uuid = uuid.UUID(user_id) if isinstance(user_id, str) else user_id
+            except ValueError:
+                self.logger.warning("Cannot hydrate strategies from DB - invalid user_id format", user_id=user_id)
+                return []
+
+            async with get_database() as db:
```

Ruff (0.13.1)

Undefined name `get_database`

(F821)